Rollbar/Telemetry/TelemetryCollector.cs

```diff
@@ -0,0 +1,275 @@
+            {
+            }
+
+            /// <summary>
+            /// The singleton-like instance of the service.
+            /// </summary>
+            internal static readonly TelemetryCollector Instance =
+                new TelemetryCollector();
+        }
+
+        #endregion singleton implementation
+
+        /// <summary>
+        /// Captures the specified telemetry.
+        /// </summary>
+        /// <param name="telemetry">The telemetry.</param>
+        /// <returns></returns>
+        public TelemetryCollector Capture(Telemetry telemetry)
+        {
+            if (this.Config.TelemetryEnabled)
+            {
+                this._telemetryQueue.Enqueue(telemetry);
+            }
+            return this;
+        }
+
+        /// <summary>
+        /// Captures the specified telemetry.
+        /// </summary>
+        /// <param name="telemetry">The telemetry.</param>
+        /// <returns></returns>
+        ITelemetryCollector ITelemetryCollector.Capture(Telemetry telemetry)
+        {
+            return this.Capture(telemetry);
+        }
+
+        /// <summary>
+        /// Gets the items count.
+        /// </summary>
+        /// <returns></returns>
+        public int GetItemsCount()
+        {
+            return this._telemetryQueue.GetItemsCount();
+        }
+
+        /// <summary>
+        /// Gets the content of the queue.
+        /// </summary>
+        /// <returns></returns>
+        public Telemetry[] GetQueueContent()
+        {
+            return this._telemetryQueue.GetQueueContent();
+        }
+
+        /// <summary>
+        /// Flushes the queue.
+        /// </summary>
+        /// <returns></returns>
+        public ITelemetryCollector FlushQueue()
+        {
+            this._telemetryQueue.Flush();
+            return this;
+        }
+
+        /// <summary>
+        /// Gets the configuration.
+        /// </summary>
+        /// <value>
+        /// The configuration.
+        /// </value>
+        public TelemetryConfig Config { get { return this._config; } }
+
+        /// <summary>
+        /// Starts the auto-collection.
+        /// </summary>
+        public ITelemetryCollector StartAutocollection()
+        {
+            if (!this.Config.TelemetryEnabled)
+            {
+                return this; // no need to start at all...
+            }
+            if (this.Config.TelemetryAutoCollectionTypes == TelemetryType.None)
+            {
+                return this; // nothing really to collect...
+            }
+            if (this.Config.TelemetryAutoCollectionInterval == TimeSpan.Zero)
+            {
+                return this; // this, essentially, means the auto-collection is off...
+            }
+
+            // let's resync with relevant config settings: 
+            this._telemetryQueue.QueueDepth = this.Config.TelemetryQueueDepth;
+
+            lock (_syncRoot)
+            {
+                if (this._telemetryThread == null)
```

Comment:
This is basically just stub code for now because the auto collection isn't doing anything yet as far as I can tell, right?

Reply:
correct. once we finalize on what to auto-collect and how, this auto-collection thread will be activated.